src/org/mozilla/javascript/VMBridge.java

```diff
@@ -10,23 +10,24 @@
 
 import java.lang.reflect.AccessibleObject;
 import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Map;
 
-public abstract class VMBridge
-{
+public abstract class VMBridge {
 
     static final VMBridge instance = makeInstance();
 
-    private static VMBridge makeInstance()
-    {
+    private static VMBridge makeInstance() {
         String[] classNames = {
             "org.mozilla.javascript.VMBridge_custom",
+            "org.mozilla.javascript.jdk11.VMBridge_jdk11",
             "org.mozilla.javascript.jdk18.VMBridge_jdk18",
         };
         for (int i = 0; i != classNames.length; ++i) {
             String className = classNames[i];
             Class<?> cl = Kit.classOrNull(className);
             if (cl != null) {
-                VMBridge bridge = (VMBridge)Kit.newInstanceOrNull(cl);
+                VMBridge bridge = (VMBridge) Kit.newInstanceOrNull(cl);
```

Comment:
I see no code that distinguish between the different JVM versions.
So also for a jdk8 the first instance that is used 
(or do we get a NoClassDefFound-error when VMBridge_jdk11 when instanttiated on jdk8)